rsconnect/api.py

```diff
@@ -587,13 +755,17 @@ def validate_app_mode(self, *args, **kwargs):
             if app_id is None:
                 # Possible redeployment - check for saved metadata.
                 # Use the saved app information unless overridden by the user.
-                app_id, existing_app_mode = app_store.resolve(connect_server.url, app_id, app_mode)
+                app_id, existing_app_mode = app_store.resolve(self.remote_server.url, app_id, app_mode)
                 logger.debug("Using app mode from app %s: %s" % (app_id, app_mode))
             elif app_id is not None:
                 # Don't read app metadata if app-id is specified. Instead, we need
-                # to get this from Connect.
-                app = get_app_info(connect_server, app_id)
-                existing_app_mode = AppModes.get_by_ordinal(app.get("app_mode", 0), True)
+                # to get this from the remote.
+                if isinstance(self.remote_server, RSConnectServer):
+                    app = get_app_info(self.remote_server, app_id)
+                    existing_app_mode = AppModes.get_by_ordinal(app.get("app_mode", 0), True)
+                else:
+                    app = get_shinyapp_info(self.remote_server, app_id)
+                    existing_app_mode = AppModes.get_by_cloud_name(app.json_data["mode"])
```

Comment:
the pattern we should follow now that we are accepting different backends is to have a default raise exception when we don't get a match, example as follows. This also allows for future expansion without needing to modify the else statement:
https://github.com/rstudio/rsconnect-python/blob/3e4b0a7ab31ba41cc5b070522be0cc4bfd2efc49/rsconnect/api.py#L411-L416